specification/machinelearningservices/resource-manager/Microsoft.MachineLearningServices/preview/2024-01-01-preview/workspaceRP.json

```diff
@@ -2071,86 +3243,249 @@
+        },
+        "type": {
+          "description": "Kind of the deployment.",
+          "minLength": 1,
+          "pattern": "[a-zA-Z0-9._]",
+          "type": "string"
+        }
+      },
+      "discriminator": "type"
+    },
+    "EndpointDeploymentResourcePropertiesBasicResource": {
+      "required": [
+        "properties"
+      ],
+      "type": "object",
+      "allOf": [
+        {
+          "$ref": "../../../../../common-types/resource-management/v3/types.json#/definitions/Resource"
+        }
+      ],
+      "properties": {
+        "properties": {
+          "$ref": "#/definitions/EndpointDeploymentResourceProperties"
+        }
       }
     },
-    "DiagnoseWorkspaceParameters": {
-      "description": "Parameters to diagnose a workspace",
+    "EndpointDeploymentResourcePropertiesBasicResourceArmPaginatedResult": {
       "type": "object",
       "properties": {
+        "nextLink": {
+          "type": "string"
+        },
         "value": {
-          "$ref": "#/definitions/DiagnoseRequestProperties"
+          "type": "array",
+          "items": {
+            "$ref": "#/definitions/EndpointDeploymentResourcePropertiesBasicResource"
+          }
         }
       }
     },
-    "EncryptionKeyVaultUpdateProperties": {
-      "required": [
-        "keyIdentifier"
-      ],
+    "EndpointKeys": {
       "type": "object",
       "properties": {
-        "keyIdentifier": {
-          "minLength": 1,
-          "pattern": "[a-zA-Z0-9_]",
+        "keys": {
+          "description": "Dictionary of Keys for the endpoint.",
+          "type": "object",
+          "x-ms-secret": true
+        }
+      }
+    },
+    "EndpointModels": {
+      "type": "object",
+      "properties": {
+        "nextLink": {
+          "description": "The link to the next page constructed using the continuationToken.  If null, there are no additional pages.",
           "type": "string"
+        },
+        "value": {
+          "description": "List of models.",
+          "type": "array",
+          "items": {}
         }
       }
     },
-    "EncryptionProperty": {
+    "EndpointResourceProperties": {
       "required": [
-        "keyVaultProperties",
-        "status"
+        "endpointType"
       ],
       "type": "object",
       "properties": {
-        "cosmosDbResourceId": {
-          "description": "The byok cosmosdb account that customer brings to store customer's data\r\nwith encryption",
+        "associatedResourceId": {
+          "description": "Byo resource id for creating the built-in model service endpoints.",
           "type": "string"
         },
-        "identity": {
-          "description": "Identity to be used with the keyVault",
-          "$ref": "#/definitions/IdentityForCmk"
-        },
-        "keyVaultProperties": {
-          "description": "KeyVault details to do the encryption",
-          "$ref": "#/definitions/KeyVaultProperties"
+        "endpointType": {
```

Comment:
>endpointType [](http://example.com/codeflow?start=9&length=12)

Use string enum for discriminator

Reply:
is this a must? the PM decide to use name like "Azure.OpenAI" as the type discriminator, which can't be considered as enum, can we get an exemption on this?

Reply:
You can use '.' in an enum value. Here's an example: [https://github.com/Azure/azure-rest-api-specs/blob/84e597d8832ae1f540a46acb388b91f33e510493/specification/adp/resource-manager/Microsoft.AutonomousDevelopmentPlatform/preview/2021-11-01-preview/adp.json#L667C13-L667C63](https://github.com/Azure/azure-rest-api-specs/blob/84e597d8832ae1f540a46acb388b91f33e510493/specification/adp/resource-manager/Microsoft.AutonomousDevelopmentPlatform/preview/2021-11-01-preview/adp.json#L667C13-L667C63)

Reply:
Updated